tests/std/tests/P0355R7_calendars_and_time_zones_hms/test.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <chrono>
+
+using namespace std;
+using namespace std::chrono;
+
+using hms_hours = hh_mm_ss<hours>;
+
+constexpr void am_pm() {
+    static_assert(noexcept(is_am(hours{})));
+    static_assert(noexcept(is_pm(hours{})));
+
+    for (hours i = 0h; i < 12h; ++i) {
+        assert(is_am(i));
+        assert(!is_pm(i));
+    }
+    for (hours i = 12h; i < 24h; ++i) {
+        assert(!is_am(i));
+        assert(is_pm(i));
+    }
+}
+
+constexpr void make12_24() {
+    static_assert(noexcept(make12(hours{})));
+    static_assert(noexcept(make24(hours{}, true)));
+
+    assert(make12(1h) == 1h);
+    assert(make12(13h) == 1h);
+    assert(make12(0h) == 12h);
+    assert(make12(12h) == 12h);
+
+    assert(make24(12h, true) == 12h);
+    assert(make24(12h, false) == 0h);
+    assert(make24(5h, true) == 17h);
+    assert(make24(5h, false) == 5h);
+}
+
+template <class D>
+constexpr auto width() {
+    return hh_mm_ss<D>::fractional_width;
+}
+template <intmax_t N, intmax_t D>
+constexpr auto width() {
+    return hh_mm_ss<duration<int, ratio<N, D>>>::fractional_width;

```

Comment:
Technically should include `<ratio>` (`<chrono>` isn't mandated to drag in the entire header).